airflow/providers/google/cloud/hooks/kubernetes_engine.py

```diff
@@ -373,10 +372,22 @@ def core_v1_client(self) -> client.CoreV1Api:
     def is_in_cluster(self) -> bool:
         return False
 
-    @staticmethod
-    def get_xcom_sidecar_container_image():
-        """Returns the xcom sidecar image that defined in the connection"""
-        return PodDefaults.SIDECAR_CONTAINER.image
```

Comment:
we don't need this because returning None gets you to the same place. (and we shouldn't be using `pod_generator_deprecated` anyway)